test/js/ui/map.test.js

```diff
@@ -593,6 +593,26 @@ test('Map', function(t) {
                 map.queryRenderedFeatures(map.project(new LngLat(360, 0)), opts);
             });
 
+            t.test('if no options, defaults to empty object', function(t) {
+                map.style.queryRenderedFeatures = function (coords, o) {
+                    t.ok(coords);
+                    t.deepEqual(o, {}, 'options are set to a default object');
+                    t.end();
+                };
+
+                map.queryRenderedFeatures(map.project(new LngLat(0, 0)));
+            });
+
+            t.test('options carry through query', function(t) {
+                map.style.queryRenderedFeatures = function (coords, o) {
+                    t.ok(coords);
+                    t.deepEqual(o, {foo: 'bar'}, 'options are expected');
+                    t.end();
+                };
+
+                map.queryRenderedFeatures(map.project(new LngLat(0, 0)), {foo: 'bar'});
+            });
+
             t.end();
```

Comment:
@lucaswoj I grabbed this from the [two](https://github.com/mapbox/mapbox-gl-js/pull/2979/files/76381412754bc117f0c6d35f030c50c49694e0cb#diff-277bb5e9f409aee234cc524db362a55bR568) tests [above](https://github.com/mapbox/mapbox-gl-js/pull/2979/files/76381412754bc117f0c6d35f030c50c49694e0cb#diff-277bb5e9f409aee234cc524db362a55bR580) without thinking much about it. Happy to find an alternative!